recipe/cmake-tests/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+cmake_minimum_required(VERSION 3.8)
+
+project(diana)
+
+option(MODERN_CUDA "Find CUDA with modern approach" ON)
+
+file(GLOB cpu_source_files "${CMAKE_SOURCE_DIR}/src/*.cc")
+file(GLOB gpu_source_files "${CMAKE_SOURCE_DIR}/src/*.cu")
+
+
+if(MODERN_CUDA)
+    # Modern CMake
+    set(CMAKE_CUDA_STANDARD 17)
+    set(CMAKE_CUDA_STANDARD_REQUIRED TRUE)
+    enable_language(CUDA)
+    add_executable(diana ${cpu_source_files} ${gpu_source_files})
+else()
+    # Old CMake
+    add_executable(diana ${cpu_source_files})
+    find_package(CUDA REQUIRED)
+    set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -std=c++17")
+    CUDA_ADD_LIBRARY(diana_gpu ${gpu_source_files})
+    target_link_libraries(diana diana_gpu)
+endif()
```

Comment:
Could we also test the `FindCUDAToolkit` approach? I.E. `find_package(CUDAToolkit)`

Ideally we should test that `FindCUDA` and `FindCUDAToolkit` both work with and without doing `enable_language(CUDA)` which changes the behavior.

Reply:
Hm, so the full matrix would be:

* FindCUDA
* FindCUDA + enable_language
* FindCUDAToolkit
* FindCUDAToolkit + enable_language

right?

Reply:
I would add just `enable_language` as well.

Reply:
_FindCUDAToolkit without enable_language_  does not seem to be a valid combination.

Reply:
> _FindCUDAToolkit without enable_language_ does not seem to be a valid combination.

It absolutely should be and I have projects that do exactly this. It sounds like we're not setting a necessary `env` var or we're doing something that causes `FindCUDAToolkit` to fail.